docs/integrations/webhooks/create-a-webhook.mdx

```diff
@@ -0,0 +1,361 @@
+    return new Response('Error occured -- no svix headers', {
+      status: 400
+    })
+  }
+
+  // Get the body
+  const payload = await req.json()
+  const body = JSON.stringify(payload);
+
+  // Create a new Svix instance with your secret.
+  const wh = new Webhook(WEBHOOK_SECRET);
+
+  let evt: WebhookEvent
+
+  // Verify the payload with the headers
+  try {
+    evt = wh.verify(body, {
+      "svix-id": svix_id,
+      "svix-timestamp": svix_timestamp,
+      "svix-signature": svix_signature,
+    }) as WebhookEvent
+  } catch (err) {
+    console.error('Error verifying webhook:', err);
+    return new Response('Error occured', {
+      status: 400
+    })
+  }
+
+  // Do something with the payload
+  // For this guide, you simply log the payload to the console
+  const { id } = evt.data;
+  const eventType = evt.type;
+  console.log(`Webhook with and ID of ${id} and type of ${eventType}`)
+  console.log('Webhook body:', body)
+
+  return new Response('', { status: 200 })
+}
+
+```
+
+```ts filename="pages/api/webhooks.ts"
+import { Webhook } from 'svix'
+import { WebhookEvent } from '@clerk/nextjs/server'
+import { NextApiRequest, NextApiResponse } from 'next'
+import { buffer } from 'micro'
+
+export const config = {
+  api: {
+    bodyParser: false,
+  }
+}
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== 'POST') {
+    return res.status(405)
+  }
+  // You can find this in the Clerk Dashboard -> Webhooks -> choose the webhook
+  const WEBHOOK_SECRET = process.env.WEBHOOK_SECRET
+
+  if (!WEBHOOK_SECRET) {
+    throw new Error('Please add WEBHOOK_SECRET from Clerk Dashboard to .env or .env.local')
+  }
+
+  // Get the headers
+  const svix_id = req.headers["svix-id"] as string;
+  const svix_timestamp = req.headers["svix-timestamp"] as string;
+  const svix_signature = req.headers["svix-signature"] as string;
+
+
+  // If there are no headers, error out
+  if (!svix_id || !svix_timestamp || !svix_signature) {
+    return res.status(400).json({ error: 'Error occured -- no svix headers' })
+  }
+
+  console.log('headers', req.headers, svix_id, svix_signature, svix_timestamp)
+  // Get the body
+  const body = (await buffer(req)).toString()
+
+  // Create a new Svix instance with your secret.
+  const wh = new Webhook(WEBHOOK_SECRET);
+
+  let evt: WebhookEvent
+
+  // Verify the payload with the headers
+  try {
+    evt = wh.verify(body, {
+      "svix-id": svix_id,
+      "svix-timestamp": svix_timestamp,
+      "svix-signature": svix_signature,
+    }) as WebhookEvent
+  } catch (err) {
+    console.error('Error verifying webhook:', err);
+    return res.status(400).json({ 'Error': err })
+  }
+
+  // Do something with the payload
+  // For this guide, you simply log the payload to the console
+  const { id } = evt.data;
+  const eventType = evt.type;
+  console.log(`Webhook with and ID of ${id} and type of ${eventType}`)
+  console.log('Webhook body:', body)
```

Comment:
```suggestion
  console.log(`Webhook with and ID of ${id} and type of ${eventType}`)
  if (evt.type === 'user.updated') {
    console.log('Data from user.updated webhook:', evt.data)
  }

```

This will scope the full logging to only the user.updated event, which is what's being used in the guide. It will also point customers in the right direction TypeScript as anything in the `if()` will now correctly infer the appropriate properties.